bids/config.py

```diff
@@ -98,8 +97,8 @@ def _update_from_standard_locations():
     current directory--with later files taking precedence over earlier ones.
     """
     locs = [
-        join(expanduser('~'), _config_name),
-        join('.', _config_name)
+        Path.home() / _config_name,
+        Path('.') / _config_name
```

Comment:
```suggestion
        Path() / _config_name
```

Reply:
This is just:
```suggestion
        Path(_config_name)
```

But IMO, the goal should be to work equally well with `Path`s and `str`s, so it shouldn't be any less valid to write:

```suggestion
        _config_name
```

Reply:
I assumed that the joining with '.' was done on purpose in the original `join('.', _config_name)`. The only case I could come up with where there would be a difference is `_config_name` starting with a `/`. I am not sure if that what the original author was going for but I didn't want to hide this change under "switching to pathlib".

Reply:
@adelavega, I am not sure what to do about this one. I don't know if the original joining with '.' was done on purpose or not so I am wary of removing it as suggested.

Reply:
I think @effigies is correct, and all version of those paths will resolve to be the same `Path(_config_name)` is equivalent to a file in the current directory.

I'm fine w/ using Pathlib internally, though @effigies so I would favor:

```
Path(_config_name)
```


Also I think the

Reply:
L105 is calling `from_file()` which is an external API that has been written to handle `str` or `Path`. Why are we insisting on converting to `Path` here?

Sorry, just in general, I think:

1) We should support both Path and str in our APIs.
2) `os.path.*` supports `Path`, so we don't need to obsessively purge it. Sometimes `pathlib` ends up being more verbose to no benefit.

Reply:
1. Agree, 
2. Sure, although I'm not too concerned about this. I could go either way. I personally like to go as far with pathlib as I can as its easier for me to read the code. No strong preference though.

Reply:
@effigies, what should I do about this one? 

I didn't use `Path('.') / _config_name` to convert `_config_name` to `Path`. Instead I used it as a `pathlib` version of the original `join('.', _config_name)`. Was this joining with `.` meaningless? If so, then we can use `_config_name` by itself. The only doubt that I have about this option is that this would be a very confusing change if done under "switching to pathlib".